### DISCUSSION TITLE
Invalid Provisioning Profile

### DISCUSSION THREAD

**Discussion by sanketWohlig** [codemagic-ci-cd/codemagic-docs#1839] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

I am unable to publish the app to ios.
I uploaded all certificate and profile in codemagic 
Codemagic file
- name: KeyChain
        script: |
          keychain initialize
          keychain add-certificates
      - name: Set up code signing settings on Xcode project
        script: |
          xcodebuild -allowProvisioningUpdates
          xcode-project use-profiles
      - name: Install pods
        script: |
          find . -name "Podfile" -execdir pod install \;
      - name: Flutter build ipa
        script: |
          flutter build ipa --release \
            --export-options-plist=/Users/builder/export_options.plist

Error
{"tool-version":"6.315.14315","tool-path":"\/Applications\/Xcode-14.3.app\/Contents\/SharedFrameworks\/ContentDeliveryServices.framework\/Versions\/A\/Frameworks\/AppStoreService.framework","os-version":"13.3.1","product-errors":[{"message":"Asset validation failed","userInfo":{"NSUnderlyingError":"Error Domain=IrisAPI Code=-19241 \"Asset validation failed\" UserInfo={status=409, detail=Invalid Provisioning Profile. The provisioning profile included in the bundle com.newsshield.io [Payload\/Runner.app] is invalid. [Missing code-signing certificate]. A Distribution Provisioning profile should be used when submitting apps to the App Store. For more information, visit the iOS Developer Portal., id=7b960cf3-5793-4878-944e-79a453948417, code=STATE_ERROR.VALIDATION_ERROR.90161, title=Asset validation failed, NSLocalizedFailureReason=Invalid Provisioning Profile. The provisioning profile included in the bundle com.newsshield.io [Payload\/Runner.app] is invalid. [Missing code-signing certificate]. A Distribution Provisioning profile should be used when submitting apps to the App Store. For more information, visit the iOS Developer Portal., NSLocalizedDescription=Asset validation failed}","NSLocalizedDescription":"Asset validation failed","iris-code":"STATE_ERROR.VALIDATION_ERROR.90161","NSLocalizedFailureReason":"Invalid Provisioning Profile. The provisioning profile included in the bundle com.newsshield.io [Payload\/Runner.app] is invalid. [Missing code-signing certificate]. A Distribution Provisioning profile should be used when submitting apps to the App Store. For more information, visit the iOS Developer Portal. (ID: 7b960cf3-5793-4878-944e-79a453948417)"},"code":90161}]}

### Expected results

The app should be published in the app store.

### Actual results

The app does not get uploaded to the app store.

### Build id (optional)

64a26c2ac31d083fe756ed35

**Comment by dtrdic** [comment, marked as answer]
hey @sanketWohlig a **Distribution** Provisioning profile should be used when submitting apps to the App Store, not Development. 
<img width="786" alt="image" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/b8bd7443-99e7-46d5-baf8-1dedc452e3bf">
https://developer.apple.com/account/resources/profiles/list

**Reply by sanketWohlig** [reply]
Yes, I uploaded the 
<img width="1171" alt="Screenshot 2023-07-05 at 6 23 31 PM" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/124883786/2b648eb1-3b2c-4760-8fbe-2b8edb496544">
App Store distribution  Provisioning profile.

**Reply by dtrdic** [reply]
Let's try removing the development cert. It's possible that the system is still reading it since both certificates have the same bundle ID.

**Reply by sanketWohlig** [reply]
Thks bro issue is resolved I forgot to change the distribution_type to app_store in codemagic.yaml file
